frontend/src/app/util/code_gen.cljs

```diff
@@ -125,11 +178,30 @@
           (filter (comp not nil?))
           (str/join "\n")))))
 
+(defn get-flex-elements [page-id shapes]
```

Comment:
Is this function repeated here for some reason? 
I'm also concerned that you are introducing react hook dependency on a pure utility function under utils